src/Data/Vec/Functional/Properties.agda

```diff
@@ -0,0 +1,265 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some Vector-related properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Vec.Functional.Properties where
+
+open import Data.Empty using (⊥-elim)
+open import Data.Fin.Base
+open import Data.Nat as ℕ
+import Data.Nat.Properties as ℕₚ
+open import Data.Product as Product using (_×_; _,_; proj₁; proj₂)
+open import Data.List as L using (List)
+import Data.List.Properties as Lₚ
+open import Data.Sum.Base as Sum using (_⊎_; inj₁; inj₂)
+open import Data.Vec as V using (Vec)
+import Data.Vec.Properties as Vₚ
+open import Data.Vec.Functional
+open import Function.Base
+open import Level using (Level)
+open import Relation.Binary as B
+open import Relation.Binary.PropositionalEquality
+open import Relation.Nullary using (Dec; does; yes; no)
+open import Relation.Nullary.Decidable using (map′)
+open import Relation.Nullary.Product using (_×-dec_)
+
+import Data.Fin.Properties as Finₚ
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+------------------------------------------------------------------------
+
+module _ {n} {xs ys : Vector A (suc n)} where
+
+  ∷-cong : head xs ≡ head ys × tail xs ≗ tail ys → xs ≗ ys
+  ∷-cong (eq , _) zero    = eq
+  ∷-cong (_ , eq) (suc i) = eq i
+
+  ∷-injective : xs ≗ ys → head xs ≡ head ys × tail xs ≗ tail ys
+  ∷-injective eq = eq zero , eq ∘ suc
+
+≗-dec : B.Decidable {A = A} _≡_ →
```

Comment:
This can be updated to use the new `DecidableEquality` from `Relation.Binary`

Reply:
Done